File: DaS.ScriptLib/Game/Data/Structures/Heading.vb
Namespace Game.Data.Structures

    Public Structure Heading

        Private _ingameValue As Single

        Public Sub New(headingAngle As Single)
            HeadingValue = headingAngle
        End Sub

        Public Property PlanarValue As Double
            Get
                Return _ingameValue
            End Get
            Set(value As Double)
                _ingameValue = CType(value, Single)
            End Set
        End Property

        'Value converted as a double to prevent any noticable loss of precision in the conversion.
        Public Property HeadingValue As Double
            Get
                Return (_ingameValue / Math.PI * 180.0) + 180.0
            End Get
            Set(value As Double)
                _ingameValue = CType((value * Math.PI / 180.0) - Math.PI, Single)
            End Set
        End Property

    End Structure

End Namespace

File: DaS.ScriptLib.NEW/Game/Data/Structures/Heading.cs
using System;
namespace DaS.ScriptLib.Game.Data.Structures
{

    public class Heading
	{
		private float _ingameValue;

        public Heading() { }

		public Heading(float headingAngle)
		{
			HeadingValue = headingAngle;
		}

		public double PlanarValue {
			get { return _ingameValue; }
			set { _ingameValue = (float)value; }
		}

		//Value converted as a double to prevent any noticable loss of precision in the conversion.
		public double HeadingValue {
			get { return (_ingameValue / Math.PI * 180.0) + 180.0; }
			set { _ingameValue = (float)((value * Math.PI / 180.0) - Math.PI); }
		}

	}

}
